.github/workflows/tests.yml

```diff
@@ -76,3 +76,25 @@ jobs:
         # XDG_RUNTIME_DIR is set.
         # TODO: figure out what exactly podman needs
         sudo -E XDG_RUNTIME_DIR= pytest-3 -s -vv
+
+  integration-macos:
+    name: "Integration macos"
+    # needed to get latest cpu
+    runs-on: macos-13
+    steps:
+    - uses: actions/checkout@v4
```

Comment:
I'd suggest using the full action version (vX.Y.Z) because there might be rare cases where action versions are not backward compatible.

Reply:
Thanks for this suggestion. I would like to get input from @ondrejbudai or @achilleas-k because AFAICT we are not doing this in any of our other repos. I'm happy to check this but want it to be consistent (in this repo at least). It will be a separate PR that changes it for all checkout actions in the workflow. Hope that is okay?

Reply:
No strong opinion here, I generally like pinning deps, but I have never seen an issue with `actions/checkout` (and our use-case is trivial), so I don't care if this particular one is pinned.